hypertrace-trace-enricher/hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/clients/DefaultClientRegistry.java

```diff
@@ -29,8 +29,9 @@
 import org.hypertrace.traceenricher.util.UserAgentParser;
 
 public class DefaultClientRegistry implements ClientRegistry {
-  private static final String ATTRIBUTE_SERVICE_HOST_KEY = "attribute.service.config.host";
-  private static final String ATTRIBUTE_SERVICE_PORT_KEY = "attribute.service.config.port";
+  private static final String ATTRIBUTE_SERVICE_CONFIG_KEY = "attribute.service.config";
+  private static final String ATTRIBUTE_SERVICE_HOST_KEY = ATTRIBUTE_SERVICE_CONFIG_KEY + ".host";
```

Comment:
They're used to build channel
```java
this.attributeServiceChannel =
        this.buildChannel(
            config.getString(ATTRIBUTE_SERVICE_HOST_KEY),
            config.getInt(ATTRIBUTE_SERVICE_PORT_KEY));
```

Reply:
Ah I thought that went away since you removed the getter and are passing in the raw config to the client. I wonder if passing in the registry would be better so it can read the channel config in the same way? Oh well, doesn't really matter - only thing I would suggest in this case would be to remove the instance variable for channel since now it's only used locally in creating the cached client.